src/QsCompiler/Transformations/ClassicallyControlled.cs

```diff
@@ -845,6 +845,25 @@ public LiftContent()
                 this.StatementKinds = new StatementKindTransformation(this);
             }
 
+            private static bool IsConditionValidForLifting(QsConditionalStatement conditionStatement)
+            {
+                if (conditionStatement.ConditionalBlocks.Length == 1)
```

Comment:
I think this is correct and will not result in a change of behavior, but it's worth digging in to make absolutely sure. 

At this point, because of previous transformations, each condition will have exactly one block. We can test this with some specific input programs to confirm.

Reply:
This is not obvious to see in the code, which makes this change a risk. We need full validation for that, and we need to double check that there is sufficient test coverage that would catch it if something in that logic does not behave as expected.

Reply:
I was able to work through this code locally, and am confident in the approach. I even tried modifying it locally and replacing the else branch with a throw, and that throw was never hit in any of the existing or new tests or any manually checked cases. I'm pretty sure that by the time we get here the `ConditionalBlocks.Length` is guaranteed to be 1, barring a bug in a previous transformation. Given that, I'd even be in favor of having the else block be a throw or using `var condition = conditionStatement.ConditionalBlocks.Single().Item1;` to ensure that the code will fail if that condition is ever violated. Otherwise, I'm fine with it as-is!

Reply:
I've update the code to use the Single LINQ function instead.